### ISSUE TITLE
X509Chain.Build throws on macOS for certificates with an empty subject

### ISSUE THREAD

**Issue by qmfrederik** [dotnet/corefx#29577]
Calling `X509Chain.Build` with a certificate which has an empty subject will return `false` on Windows and Linux but throw a `CryptographicException` on macOS:

```
System.Security.Cryptography.CryptographicException : Error occurred during a cryptographic operation.
Stack Trace:
     at Internal.Cryptography.Pal.SecTrustChainPal.ParseResults(SafeX509ChainHandle chainHandle, X509RevocationMode revocationMode)
     at Internal.Cryptography.Pal.SecTrustChainPal.Execute(DateTime verificationTime, Boolean allowNetwork, OidCollection applicationPolicy, OidCollection certificatePolicy, X509RevocationFlag revocationFlag)
     at Internal.Cryptography.Pal.ChainPal.BuildChain(Boolean useMachineContext, ICertificatePal cert, X509Certificate2Collection extraStore, OidCollection applicationPolicy, OidCollection certificatePolicy, X509RevocationMode revocationMode, X509RevocationFlag revocationFlag, DateTime verificationTime, TimeSpan timeout)
     at System.Security.Cryptography.X509Certificates.X509Chain.Build(X509Certificate2 certificate, Boolean throwOnException)
     at System.Security.Cryptography.X509Certificates.X509Chain.Build(X509Certificate2 certificate)
```

**Comment by qmfrederik** [comment]
This _seems_ to happen because macOS returns an error `NonEmptySubject`, which is not handled by  [MergeStatusCodes](https://github.com/dotnet/corefx/blob/master/src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509chain.cpp#L129).

Couple of fun facts:
- This value, `NonEmptySubject`, is defined as `kSecPolicyCheckNonEmptySubject` in the [open source code for Security](https://opensource.apple.com/source/Security/Security-55471/sec/Security/SecPolicy.c), but it is not in the public headers.
- .NET Core appears to rely on the result of `SecTrustEvaluate` and `SecTrustCopyResult `, though the Apple documentation explicitly states that ["The format and contents of the trust exception data is not considered API. This technique is for debugging purposes only."](https://developer.apple.com/library/content/technotes/tn2232/_index.html#//apple_ref/doc/uid/DTS40012884-CH1-SECINVESTIGATINGTRUSTEVALUATIONFAILURES)

So the fix appears to be to add `NonEmptySubject` to the list of values in `MergeStatusCodes`?

**Comment by karelz** [comment]
Fixed in PR #30427 in 2.1 servicing - it will be part of 2.1.3 release. Closing.
